### PR TITLE
set listener mask on set_listener [8787]

### PR DESCRIPTION
**Pull request by IkerLuengo**
- New set_listener that takes a mask as argument
- Testing for the mask

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2240/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2047/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2239/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2242/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2049/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2241/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2251/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2058/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2250/)

File: include/fastdds/dds/domain/DomainParticipant.hpp [hunk 1]
```diff
@@ -109,13 +109,23 @@ class DomainParticipant : public Entity
...
+     * @param listener new value for the DomainParticipantListener
      * @return RETCODE_OK
      */
     RTPS_DllAPI ReturnCode_t set_listener(
             DomainParticipantListener* listener);
 
+    /**
+     * Modifies the DomainParticipantListener.
+     * @param listener new value for the DomainParticipantListener
+     * @param mask StatusMask
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipant.hpp:122]: Improve the documentation of the `mask` parameter.

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 2]
```diff
@@ -231,13 +231,23 @@ class DataWriter : public DomainEntity
...
+     * @param listener new value for the DataWriterListener
      * @return RETCODE_OK
      */
     RTPS_DllAPI ReturnCode_t set_listener(
             DataWriterListener* listener);
 
+    /**
+     * Modifies the DataWriterListener.
+     * @param listener new value for the DataWriterListener
+     * @param mask StatusMask
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:244]: Improve the documentation of the `mask` parameter.

File: include/fastdds/dds/publisher/Publisher.hpp [hunk 3]
```diff
@@ -120,13 +120,23 @@ class Publisher : public DomainEntity
...
+     * @param listener new value for the PublisherListener
      * @return RETCODE_OK
      */
     RTPS_DllAPI ReturnCode_t set_listener(
             PublisherListener* listener);
 
+    /**
+     * Modifies the PublisherListener.
+     * @param listener new value for the PublisherListener
+     * @param mask StatusMask
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/Publisher.hpp:133]: Improve the documentation of the `mask` parameter.

File: include/fastdds/dds/subscriber/DataReader.hpp [hunk 4]
```diff
@@ -246,13 +246,22 @@ class DataReader : public DomainEntity
...
      * @param listener new value for the DataReaderListener
      * @return RETCODE_OK
      */
     RTPS_DllAPI ReturnCode_t set_listener(
             DataReaderListener* listener);
 
+    /**
+     * Modifies the DataReaderListener.
+     * @param listener new value for the DataReaderListener
+     * @param mask StatusMask
```
Comment (MiguelCompany) [inline include/fastdds/dds/subscriber/DataReader.hpp:259]: Improve the documentation of the `mask` parameter.

File: include/fastdds/dds/subscriber/Subscriber.hpp [hunk 5]
```diff
@@ -127,13 +127,22 @@ class Subscriber : public DomainEntity
...
      * @param listener new value for SubscriberListener
      * @return RETCODE_OK
      */
     RTPS_DllAPI ReturnCode_t set_listener(
             SubscriberListener* listener);
 
+    /**
+     * Modifies the SubscriberListener.
+     * @param listener new value for the SubscriberListener
+     * @param mask StatusMask
```
Comment (MiguelCompany) [inline include/fastdds/dds/subscriber/Subscriber.hpp:140]: Improve the documentation of the `mask` parameter.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2259/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2061/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2253/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2264/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2065/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2257/)

**Pull request merged**